### PR TITLE
feat(VBtn): support Material Design 2

### PR DESCRIPTION
**Pull request by sh7dm**
## Description
Make VBtn compatible with md2 spec. **Don't squash!**

## Motivation and Context
Vuetify 2.0 should be compatible with md2.

## How Has This Been Tested?
Visually + `jest`

## Markup:
<details open>

Use the **v-btn** kitchen.

```bash
yarn kitchen
```
</details>

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library) 

## 2.0 Tasks:
- [x] Convert to TS (if applicable)
- [x] Convert Stylus to SCSS
- [x] Convert tests
- [x] Update component folder
  - [x] Migrate styles
  - [x] Migrate tests
- [x] Define updates needed to convert to md2
- [x] Update component to md2 spec
- [ ] Check WAI-ARIA compatibility
- [x] Fix icons
default

## 2.0 Misc. Tasks:
- [ ] Kitchen Sink
- [ ] [Global SCSS Structure Definition [WIP]](https://gist.github.com/MajesticPotatoe/35def600403a0cf4b1df94f6e95cce41) - @MajesticPotatoe 

## Playground Components
- [x] Principles
  - [x] Identifiable
  - [x] Findable
  - [x] Clear
- [x] Types: all of types are present + Vuetify has even more types of buttons
  - [x] Text button
  - [x] Outlined button
  - [x] Contained button
  - [x] Toggle button
- [x] Should match "anatomy" examples
  - [x] Text button
  - [x] Outlined button
  - [x] Contained button
  - [x] Toggle button
- [x] States
  - [x] Enabled
    - [x] Text button
    - [x] Outlined button
    - [x] Contained button
    - [x] Toggle button
  - [x] Disabled
    - [x] Text button
    - [x] Outlined button
    - [x] Contained button
    - [x] Toggle button
  - [x] Hover
    - [x] Text button
    - [x] Outlined button
    - [x] Contained button
    - [x] Toggle button
  - [x] Focused
    - [x] Text button
    - [x] Outlined button
    - [x] Contained button
    - [x] Toggle button
  - [x] Pressed
    - [x] Text button
    - [x] Outlined button
    - [x] Contained button
    - [x] Toggle button

### CODE REVIEW

File: packages/vuetify/src/scss/variables/_colors.scss [hunk 1]
```diff
@@ -0,0 +1,320 @@
+$red: (
+  'base': #F44336,
+  'lighten-5': #FFEBEE,
```
Comment (sh7dm) [inline packages/vuetify/src/scss/variables/_colors.scss]: Maybe use scss `lighten`, `darken` and `saturate` functions to compute these colors?

Comment (KaelWD) [reply packages/vuetify/src/scss/variables/_colors.scss]: https://material.io/design/color/the-color-system.html#color-usage-palettes

I don't know if there's a particular pattern behind them, even material-components-web hardcodes them. It doesn't matter anyway as they aren't intended to be changed.

Comment (sh7dm) [reply packages/vuetify/src/scss/variables/_colors.scss]: @KaelWD okay

File: packages/vuetify/build/webpack.base.config.js [hunk 2]
```diff
@@ -50,7 +50,11 @@ exports.config = {
       },
       {
         test: /\.scss$/,
-        use: scssLoaders
+        use: sassLoaders
```
Comment (sh7dm) [inline packages/vuetify/build/webpack.base.config.js]: We can do simplier: `/\.s(a|c)ss$/`

File: packages/vuetify/src/components/VBtn/VBtn.sass [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+.v-btn--fixed
+  position: fixed
+
+.v-btn--icon
+  .v-icon
+    font-size: 18px
+
+.v-btn--outline
+  border: 2px solid currentColor
```
[the remark this answers is not in the document]
Comment (sh7dm) [reply packages/vuetify/src/components/VBtn/VBtn.sass]: @nekosaur like this? 
default

File: packages/vuetify/src/components/VBtn/VBtn.styl [hunk 4]
```diff
@@ -1,6 +1,6 @@
-@import '../bootstrap'
-@import '../components/_ripples'
-@import '../theme'
+@import '../../stylus/bootstrap'
+@import '../../stylus/components/_ripples'
+@import '../../stylus/theme'
```
Comment (sh7dm) [inline packages/vuetify/src/components/VBtn/VBtn.styl]: I think we can remove this file at all

File: packages/vuetify/src/components/VBtn/VBtn.sass (hunk 3 above)
[the remark this answers is not in the document]
Comment (nekosaur) [reply packages/vuetify/src/components/VBtn/VBtn.sass]: Yes. The color should probably be put in a variable though instead of being hard-coded

**Comment by chanlito** [comment]
VBtn `outline` is not the same as shown here https://material.io/design/components/buttons.html#outlined-button

**Pull request merged**